lib/mongo_ecto.ex

```diff
@@ -508,55 +509,73 @@ defmodule Mongo.Ecto do
 
     {:ok, datetime}
   end
-  defp dump_naive_datetime(_),
-    do: :error
-
-  @doc """
-  Copy from the Elixir 1.4.5. TODO: Replace with native methods, when we stick on ~> 1.4.
-  Source: https://github.com/elixir-lang/elixir/blob/v1.4/lib/elixir/lib/calendar.ex#L1477
-  """
-  defp datetime_from_naive(%NaiveDateTime{hour: hour, minute: minute, second: second, microsecond: microsecond,
-                                year: year, month: month, day: day}, "Etc/UTC") do
-    {:ok, %DateTime{year: year, month: month, day: day,
-                    hour: hour, minute: minute, second: second, microsecond: microsecond,
-                    std_offset: 0, utc_offset: 0, zone_abbr: "UTC", time_zone: "Etc/UTC"}}
-  end
 
-  @doc """
-  Copy from the Elixir 1.4.5. TODO: Replace with native methods, when we stick on ~> 1.4.
-  Source: https://github.com/elixir-lang/elixir/blob/v1.4/lib/elixir/lib/calendar.ex#L1477
-  """
+  defp dump_naive_datetime(_), do: :error
+
+  # Copy from the Elixir 1.4.5. TODO: Replace with native methods, when we stick on ~> 1.4.
+  # Source: https://github.com/elixir-lang/elixir/blob/v1.4/lib/elixir/lib/calendar.ex#L1477
+  defp datetime_from_naive(
```

Comment:
this look weird, we probably should destructure the NaiveDateTime within the function rather than in the function header.

Reply:
This will be gone soonish. I would prefer just to leave it for now.